test/lit/fuzz-types/nominal.test

```diff
@@ -0,0 +1,24 @@
+;; RUN: wasm-fuzz-types --nominal -v --seed=0
+
+;; CHECK:      Running with seed 0
+;; CHECK-NEXT: Built 20 types:
+;; CHECK-NEXT: 0: (struct (field (ref any) i8))
+;; CHECK-NEXT: 1: (array (mut i16))
+;; CHECK-NEXT: 2: (func (result anyref))
+;; CHECK-NEXT: 3: (func (result anyref))
+;; CHECK-NEXT: 4: (struct (field (ref any) i8))
+;; CHECK-NEXT: 5: (func (result anyref))
+;; CHECK-NEXT: 6: (struct (field (ref any) i8))
+;; CHECK-NEXT: 7: (struct (field (ref (array (mut i16))) i8 f32))
+;; CHECK-NEXT: 8: (struct (field (ref any) i8))
+;; CHECK-NEXT: 9: (func (result anyref))
+;; CHECK-NEXT: 10: (struct (field (ref eq) i8 (rtt func)))
+;; CHECK-NEXT: 11: (array (mut i16))
+;; CHECK-NEXT: 12: (struct (field (ref (array (mut i16))) i8 f32 (mut i8) (mut (rtt (struct (field (ref any) i8)))) v128))
+;; CHECK-NEXT: 13: (struct (field i31ref i8 (mut (ref func))))
+;; CHECK-NEXT: 14: (func (result anyref))
+;; CHECK-NEXT: 15: (struct (field (ref any) i8))
+;; CHECK-NEXT: 16: (func (result anyref))
+;; CHECK-NEXT: 17: (array (mut i16))
+;; CHECK-NEXT: 18: (array (mut i16))
+;; CHECK-NEXT: 19: (struct (field (ref (array (mut i16))) i8 f32 (mut (rtt i31))))
```

Comment:
Perhaps this could also print the index of the super, if there is one?

Reply:
Yes, these tests make it clear that there is room for improvement when printing groups of interrelated HeapTypes. Do you mind if we leave improvements there to a follow-on PR?